include/ua_plugin_historical_access.h

```diff
@@ -0,0 +1,148 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
+ *
+ *    Copyright 2018 (c) basysKom GmbH <opensource@basyskom.com> (Author: Peter Rustler)
+ *    Copyright 2018 (c) Fraunhofer IOSB (Author: Julius Pfrommer)
+ */
+
+#ifndef UA_PLUGIN_HISTORICAL_ACCESS_H_
+#define UA_PLUGIN_HISTORICAL_ACCESS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_server.h"
+
+/**
+ * .. _historical-access:
+ *
+ * Historical Access Plugin API
+ * ============================
+ *
+ * Storage and access to historical data is implemented in a user-defined
+ * plugin. The plugin API does not mandate a specific kind of storage backend or
+ * handling of continuation points.
+ *
+ * The plugin is set globally in the server and used by the HistoryRead and
+ * HistoryUpdate services.
+ *
+ * The function pointers in the HistoryAccess plugin can be set to NULL. The
+ * server returns ``UA_STATUSCODE_BADNOTIMPLEMENTED`` if a requested method is
+ * not available.
+ */
+
+typedef struct {
+    void *context;
+    void (*deleteMembers)(void *context);
+
+    /* Is the node historizing? */
+    UA_Boolean (*isHistorizing)(UA_Server *server, void *haContext,
+                                const UA_NodeId *nodeId);
+
+    /***************/
+    /* HistoryRead */
+    /***************/
+
+    UA_StatusCode
+    (*historyRead_raw)(UA_Server *server, void *haContext,
```

Comment:
haContext is not needed, because it is accessable from the server->config object.

Reply:
The server object is intentionally opaque for the end user.
The above is possible only when using internal headers.